src/main/scala/shade/memcached/internals/SpyMemcachedIntegration.scala

```diff
@@ -354,7 +354,16 @@ class SpyMemcachedIntegration(cf: ConnectionFactory, addrs: Seq[InetSocketAddres
     prepareFuture(key, op, promise, timeout)
   }
 
-  protected final def prepareFuture[T](key: String, op: Operation, promise: Promise[Result[T]], atMost: FiniteDuration)(implicit ec: ExecutionContext): Future[Result[T]] = {
+  protected final def prepareFuture[T](key: String, op: Operation, promise: Promise[Result[T]], atMost: FiniteDuration)(implicit ec: ExecutionContext): CancelableFuture[Result[T]] = {
+    val operationCancelable = Cancelable(() => {
+      try {
+        if (!op.isCancelled)
+          op.cancel()
+      } catch {
+        case NonFatal(_) =>
```

Comment:
Catching errors silently isn't a good thing to do. Even if I did it myself in the past, it was a bad thing to do :-) 

Report the error by means of `ec.reportError(ex)`, which should log it.